### PR TITLE
Bulk insert with random column list

### PR DESCRIPTION
**Pull request by balazstbb**
The usage of `useBulkCopyForBatchInsert=true` connection parameter leads to invalid batch inserts.

Requirements of the problem:
- `useBulkCopyForBatchInsert` connection parameter value is `true`
- the statement and the data types are compatible with the useBulkCopyForBatchInsert requirements [known-limitations](https://learn.microsoft.com/en-us/sql/connect/jdbc/use-bulk-copy-api-batch-insert-operation?view=sql-server-ver16#known-limitations)
- the columns provided in the insert in a different order then defined in the table schema

This PR contains a JUnit test
`com.microsoft.sqlserver.jdbc.preparedStatement.BatchExecutionWithBulkCopyTest.testReverseColumnOrder()`
that can be used to reproduce the problem.

This fix takes the column list order make sure the right values picked for the columns.

Issue: https://github.com/microsoft/mssql-jdbc/issues/1993

### CODE REVIEW

**Comment by tkyc** [comment]
We'll take a look at the PR after we complete our hotfix releases. Thanks for the contribution.

File: src/test/java/com/microsoft/sqlserver/jdbc/preparedStatement/BatchExecutionWithBulkCopyTest.java [hunk 1]
```diff
@@ -721,6 +721,45 @@ public void testNonSupportedColumns() throws Exception {
...
     }
 
+    @Test
+    public void testReverseColumnOrder() throws Exception {
+        String valid = "insert into " + AbstractSQLGenerator.escapeIdentifier(tableName) + " (c2, c1) values "
+                + "(" + "?, " + "? " + ")";
+
+        try (Connection connection = PrepUtil.getConnection(connectionString + ";useBulkCopyForBatchInsert=true;");
+                SQLServerPreparedStatement pstmt = (SQLServerPreparedStatement) connection.prepareStatement(valid);
+                Statement stmt = (SQLServerStatement) connection.createStatement();) {
```
Comment (Jeffery-Wasty) [inline src/test/java/com/microsoft/sqlserver/jdbc/preparedStatement/BatchExecutionWithBulkCopyTest.java:731]: Unneeded cast and unnecessary semicolon.

```suggestion
                Statement stmt = connection.createStatement()) {
```

File: src/test/java/com/microsoft/sqlserver/jdbc/preparedStatement/BatchExecutionWithBulkCopyTest.java [hunk 2]
```diff
@@ -721,6 +721,45 @@ public void testNonSupportedColumns() throws Exception {
         }
     }
 
+    @Test
+    public void testReverseColumnOrder() throws Exception {
+        String valid = "insert into " + AbstractSQLGenerator.escapeIdentifier(tableName) + " (c2, c1) values "
```
[the remark this answers is not in the document]
Comment (lilgreenbird) [reply src/test/java/com/microsoft/sqlserver/jdbc/preparedStatement/BatchExecutionWithBulkCopyTest.java:726]: this just needs to be run thru formatter

[the remark this answers is not in the document]
Comment (Jeffery-Wasty) [reply src/test/java/com/microsoft/sqlserver/jdbc/preparedStatement/BatchExecutionWithBulkCopyTest.java:726]: This will be resolved by formatting.

**Review by tkyc — APPROVED** [review]

**Review by Jeffery-Wasty — APPROVED** [review]

**Review by lilgreenbird — APPROVED** [review]

**Pull request merged**